src/plugins/bookmark/bookmark.js

```diff
@@ -18,48 +22,51 @@
  * Copyright 2016-2017 Henrik Ingo (@henrikingo)
  * Released under the MIT license.
  */
-/* global window, document, impress */
+/* global document, impress */
 
-( function( document, window ) {
+( function( document ) {
     "use strict";
-    var api;
     var hotkeys = {};
-    function hotkeyDest(event) {
-	return (hotkeys.hasOwnProperty(event.key)  ? hotkeys[event.key] :
-		hotkeys.hasOwnProperty(event.code) ? hotkeys[event.code] : null) }
-    
+    function hotkeyDest( event ) {
+	return ( hotkeys.hasOwnProperty( event.key )  ? hotkeys[ event.key ] :
+		 hotkeys.hasOwnProperty( event.code ) ? hotkeys[ event.code ] : null ); }
+
     // In pre-init phase, build a map of bookmark hotkey to div id, by reviewing all steps
     impress.addPreInitPlugin( function( root, api ) {
-	root.querySelectorAll( ".step" ).forEach(function(div) {
-            if ( div.dataset.bookmarkKeyList !== undefined && div.id != undefined) {
-		div.dataset.bookmarkKeyList.split( " " ).forEach( (k) => {
-		    if (hotkeys.hasOwnProperty(k)) hotkeys[k].push( div.id ); // if hotkeys collide, we'll cycle through the targets
-		    else hotkeys[k] = [div.id]; } ); } });
+	root.querySelectorAll( ".step" ).forEach( function( div ) {
+            if ( div.dataset.bookmarkKeyList !== undefined && div.id !== undefined ) {
+		div.dataset.bookmarkKeyList.split( " " ).forEach( ( k ) => {
+		    if ( hotkeys.hasOwnProperty( k ) ) {
+			hotkeys[ k ].push( div.id );
+		    } else { hotkeys[ k ] = [ div.id ]; } } ); } } );
```

Comment:
Can you move those closing brackets to separate lines? This makes the code more legible for beginners, imo. Also some explanation might be needed for the code here, if someone with less coding (well js, anyway) knowledge than you have reads this code.

Reply:
lol i'm surprised the linter even let me do that … haha

I don't really think there's anything to be gained from adding a bunch of punctuation over multiple lines…  my preference is to keep things as short as possible so you can see everything on one screen. They used to say "you can write Fortran in any language", now I suppose they should say "you can write Python in any language" 😺

Reply:
The linter is not properly configured for that. I have an issue open to talk about moving to eslint and how to configure it